src/NuGet.Clients/NuGet.PackageManagement.UI/Models/PackageItemListViewModel.cs

```
@@ -382,6 +470,255 @@ public bool IsPackageDeprecated
+                        BitmapStatus = IconBitmapStatus.EmbeddedIcon;
+                    }
+                    else // we cannot use the reader object
+                    {
+                        IconBitmap = Images.DefaultPackageIcon;
+                        BitmapStatus = IconBitmapStatus.DefaultIconDueToNoPackageReader;
+                    }
+                }
+            }
+            catch (Exception ex) when (IsHandleableBitmapEncodingException(ex))
+            {
+                IconBitmap = Images.DefaultPackageIcon;
+                BitmapStatus = IconBitmapStatus.DefaultIconDueToDecodingError;
+            }
+        }
+
+        private static bool IsHandleableBitmapEncodingException(Exception ex)
+        {
+            return ex is ArgumentException ||
+                ex is COMException ||
+                ex is FileFormatException ||
+                ex is InvalidOperationException ||
+                ex is NotSupportedException ||
+                ex is OutOfMemoryException ||
+                ex is IOException ||
+                ex is UnauthorizedAccessException;
+        }
+
+        private async Task FetchImageAsync(Uri iconUrl)
+        {
+            if (iconUrl is null)
+            {
+                return;
+            }
+
+            using (Stream stream = await GetStream(iconUrl))
+            {
+                if (stream != null)
+                {
+                    var iconBitmapImage = new BitmapImage();
+                    iconBitmapImage.BeginInit();
+                    iconBitmapImage.StreamSource = stream;
+
+                    try
+                    {
+                        FinalizeBitmapImage(iconBitmapImage);
+                        iconBitmapImage.Freeze();
+                        IconBitmap = iconBitmapImage;
+                        BitmapStatus = IconBitmapStatus.DownloadedIcon;
+                    }
+                    catch (Exception ex) when (IsHandleableBitmapEncodingException(ex))
```

What about when `!Handleable` ? If there's no other possibility, then we don't need this check at all. If there's another reasonably possible exception type, I think the Status is a little off (it may remain as Fetching forever; see my other comment).
One thought... maybe a `DefaultIconDueToUnknownError` to catch everything else?

---

yes, but at this point, that isn't really bad. just keeps showing the default icon forever. we used to catch all exceptions which was bad practice. i've now added the ~8 exceptions we might get from icon decoding per a nice blog post: https://faithlife.codes/blog/2010/07/exceptions_thrown_by_bitmapimage_and_bitmapframe/

i propose we leave it Fetching for every and don't catch all exceptions...but we can grow that list.

---

(i have a little change to the control template that visualizes the bitmap status...which i'd like to figure out how to check in as a little utility that we can use ... have to figure out how that should be done -- but in that case, if it showed fetching forever, that is bad....and we should use that clue to investigate.

---

My understanding is "it's Fetching forever, but we don't care right now". To be addressed in https://github.com/NuGet/Home/issues/10122